tests/tango/test_tango_signals.py

```diff
@@ -275,85 +108,53 @@ async def make_backend(
     return backend
 
 
-# --------------------------------------------------------------------
-async def prepare_device(echo_device: str, pv: str, put_value: T) -> None:
-    proxy = await DeviceProxy(echo_device)
-    setattr(proxy, pv, put_value)
-
-
-# --------------------------------------------------------------------
-class MonitorQueue:
-    """For monitoring updates in tests."""
-
-    def __init__(self, backend: SignalBackend):
-        self.updates: asyncio.Queue[Reading] = asyncio.Queue()
-        self.backend = backend
-        self.subscription = backend.set_callback(self.updates.put_nowait)
-
-    async def assert_updates(self, expected_value):
-        expected_reading = {
-            "timestamp": pytest.approx(time.time(), rel=0.1),
-            "alarm_severity": 0,
-        }
-        update_reading = dict(await asyncio.wait_for(self.updates.get(), timeout=5))
-        update_value = update_reading.pop("value")
-        assert_close(update_value, expected_value)
-        backend_reading = dict(
-            await asyncio.wait_for(self.backend.get_reading(), timeout=5)
-        )
-        backend_reading.pop("value")
-        backend_value = await asyncio.wait_for(self.backend.get_value(), timeout=5)
-        assert_close(backend_value, expected_value)
-        assert update_reading == expected_reading == backend_reading
-
-    def close(self):
-        self.backend.set_callback(None)
-
-
 # --------------------------------------------------------------------
 async def assert_monitor_then_put(
-    echo_device: str,
+    trl: str,
     pv: str,
     initial_value: T,
     put_value: T,
     descriptor: dict,
     datatype: type[T] | None = None,
 ):
-    await prepare_device(echo_device, pv, initial_value)
-    source = echo_device + "/" + pv
-    backend = await make_backend(datatype, source, allow_events=True)
+    source = get_full_attr_trl(trl, pv)
+    signal = tango_signal_rw(datatype, source)
+    backend = signal._connector.backend
+    await signal.connect()
+    converter = signal._connector.backend.converter  # type: ignore
+    converted_put = converter.value(put_value)
```

Comment:
Can we get these from the everything device rather than making the signals directly here?